tobac/utils/internal/xarray_utils.py

```diff
@@ -41,10 +173,252 @@ def find_vertical_axis_from_coord(
+        time_dim_number is None and ndims == 3
+    )
+    vdim_coord = None
+    if is_3d:
+        hdim1_axis, hdim2_axis, vertical_axis = find_hdim_axes_3d(
+            variable_da,
+            vertical_coord,
+            vertical_axis,
+            time_dim_coord_name=time_dim_name,
+            return_vertical_axis=True,
+        )
+        if vertical_axis is None:
+            vdim_coord = find_vertical_coord_name(variable_da, vertical_coord)
+            vertical_axis = find_axis_from_dim_coord(variable_da, vdim_coord)
+        else:
+            vdim_coord = variable_da.dims[vertical_axis]
+    else:  # 2D
+        if ndims == 2:
+            hdim1_axis = 0
+            hdim2_axis = 1
+        elif ndims == 3 and time_dim_number is not None:
+            possible_dims = [0, 1, 2]
+            possible_dims.pop(time_dim_number)
+            hdim1_axis, hdim2_axis = possible_dims
+        else:
+            raise ValueError("DataArray has too many or too few dimensions")
+
+    # If the dimensions share a name with the coordinates and those coordinates do not match
+    # with the i, j, k-style indices, you cannot `interp` along those i, j, k indices.
+    # so, instead, we rename the dimensions to random strings so that we can
+    # run interpolation.
+
+    hdim1_name_original = variable_da.dims[hdim1_axis]
+    hdim2_name_original = variable_da.dims[hdim2_axis]
+
+    # generate random names for the new coordinates that are based on i, j, k values
+    hdim1_name_new = "".join(
+        random.choice(string.ascii_uppercase + string.ascii_lowercase + string.digits)
+        for _ in range(16)
+    )
+    hdim2_name_new = "".join(
+        random.choice(string.ascii_uppercase + string.ascii_lowercase + string.digits)
+        for _ in range(16)
+    )
+
+    dim_new_names = {
+        hdim1_name_original: hdim1_name_new,
+        hdim2_name_original: hdim2_name_new,
+    }
+    dim_interp_coords = {
+        hdim1_name_new: xr.DataArray(return_feat_df["hdim_1"].values, dims="features"),
+        hdim2_name_new: xr.DataArray(return_feat_df["hdim_2"].values, dims="features"),
+    }
+
+    if is_3d:
+        vdim_name_original = variable_da.dims[vertical_axis]
+        vdim_name_new = "".join(
+            random.choice(
+                string.ascii_uppercase + string.ascii_lowercase + string.digits
+            )
+            for _ in range(16)
+        )
+        dim_interp_coords[vdim_name_new] = xr.DataArray(
+            return_feat_df["vdim"].values, dims="features"
+        )
+
+        dim_new_names[vdim_name_original] = vdim_name_new
+
+    # you can only rename dims alone when operating on datasets, so add our dataarray to a
+    # dataset
+    renamed_dim_da = variable_da.swap_dims(dim_new_names)
+    interpolated_df = renamed_dim_da.interp(coords=dim_interp_coords)
+    interpolated_df = interpolated_df.drop([hdim1_name_new, hdim2_name_new])
+    if is_3d:
+        interpolated_df = interpolated_df.drop([vdim_name_new])
```

Comment:
Suggestion: use `drop_vars` instead of the depreciated `drop` here, and use `errors=ignore` to avoid the if statement:

```suggestion
    interpolated_df = interpolated_df.drop_vars([hdim1_name_new, hdim2_name_new, vdim_name_new], errors="ignore")
```

Reply:
Agreed. I've refactored this.